server/channels/app/post_metadata.go

```diff
@@ -196,7 +196,17 @@ func (a *App) sanitizePostMetadataForUserAndChannel(c request.CTX, post *model.P
 	}
 
 	if previewedChannel != nil && !a.HasPermissionToReadChannel(c, userID, previewedChannel) {
-		post.Metadata.Embeds[0].Data = nil
+		// Remove all permalink embeds and only keep non-permalink embeds.
+		// We always have only one permalink embed even if the post
+		// contains multiple permalinks.
+		var newEmbeds []*model.PostEmbed
+		for _, embed := range post.Metadata.Embeds {
+			if embed.Type != model.PostEmbedPermalink {
+				newEmbeds = append(newEmbeds, embed)
+			}
+		}
+
+		post.Metadata.Embeds = newEmbeds
```

Comment:
That comment is actually a bit misleading as well. If I remember right, a post will only ever have a single embed currently, but I made it an array in case we ever decided to change that